node_modules/oae-principals/lib/internal/dao.js

```diff
@@ -211,6 +210,86 @@ var updatePrincipal = module.exports.updatePrincipal = function(principalId, pro
 };
 
 /**
+ * Marks a principal as being deleted
+ *
+ * @param  {String}     principalId     The id of the principal to mark as deleted
+ * @param  {Function}   callback        Standard callback function
+ * @param  {Object}     callback.err    An error that occurred, if any
+ */
+var deletePrincipal = module.exports.deletePrincipal = function(principalId, callback) {
+    var deleted = Date.now();
+
+    // Determine if the principal was already deleted in the authz index before we set them and flip
+    // the principals deleted flag
+    AuthzDelete.isDeleted([principalId], function(err, wasDeleted) {
+        if (err) {
+            return callback(err);
+        }
+
```

Comment:
No we don't. We aren't avoiding re-running the `deletePrincipal` method if the principal was already deleted, we're only keeping track in case the query to set `deleted=true` below fails, we can undo the delete if necessary.